lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4216,11 +4216,15 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
   }
 
   SILBasicBlock::iterator getNextFunctionLocalAllocationInsertionPoint() {
```

Comment:
Might be better to turn this into a function that allocates a local buffer directly: ‘createFunctionLocalBuffer(SILType)’.